src/include/miopen/tensor.hpp

```diff
@@ -182,6 +185,8 @@ struct TensorDescriptor : miopenTensorDescriptor
     bool packed;
 
     miopenDataType_t type = miopenFloat;
+
+    std::string layout;
```

Comment:
We should not be storing the layout as a string. The layout comes from the order of strides in the tensor descriptor.